### PR TITLE
04 criteria query

### PR DESCRIPTION
**Pull request by anastasiia-fi**

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -47,4 +47,9 @@ public void delete(@PathVariable Long id) {
     public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {
         return bookService.updateById(id, requestDto);
     }
+
+    @GetMapping("/search")
+    public List<BookDto> searchBooks(dto.BookSearchParametersDto searchParameters) {
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/controller/BookController.java:52]: `dto.BookSearchParametersDto` ?

this is weird :)

should be `BookSearchParametersDto`

File: src/main/java/com/example/bookstore/dao/SpecificationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.example.bookstore.dao;
+
+import dto.BookSearchParametersDto;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/dao/SpecificationBuilder.java:7]: this annotation is redundant on interface

File: src/main/java/com/example/bookstore/dto/BookSearchParametersDto.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package dto;
```
Comment (NazarSmith228) [inline src/main/java/com/example/bookstore/dto/BookSearchParametersDto.java:1]: this should not be a separate package - it should be a sub package

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

**Review by liliia-ponomarenko — APPROVED** [review]
Great 😊

**Pull request merged**